### ISSUE TITLE
App freezing while uploading a Image.

### ISSUE THREAD

**Issue by kpavankotesh** [transloadit/ios-sdk#14]
Hey, 

I am using the example described in the Readme and using a trial account to test image upload. When I select an image from my image picker in ios and pass it to transload it the app freezes and I get back the control only after I get the response from transload it. Can you please let me know how to fix this?

**Comment by Asbat** [comment]
Hi, there,

I saw the same behaviour and went with calling transloadit on a background thread, processing the response and then calling the completion block on the main thread:

`dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), {() -> Void in
    let response = transloadit?.invokeAssembly(assemblyBuilder, withError: error)
    // do something with the response

    // update your UI on the main thread
    dispatch_async(dispatch_get_main_queue(), {() -> Void in
        // update
    })
 });`

**Comment by MMasterson** [comment]
Hey @Asbat @kpavankotesh, I'll be taking a look shortly! Have you used any other SDK with your accounts before?

**Comment by kpavankotesh** [comment]
Hey, 

By `with your accounts` you mean Transloadit account? Answer: None
I am using Facebook SDK in my application.

**Comment by kpavankotesh** [comment]
Yeah that is not an error. My uploads are going through. I just pasted the logs for reference. Do let me know when you release the fix. thanks for the quick response.

**Comment by kpavankotesh** [comment]
@MMasterson I have one more doubt regarding how to read the error message in swift? I am not able to pass an NSError instance to the `invokeAssembly(self.assembly, withError: nil)` method, for now I am passing nil. 
 Can you please share a sample code or something so that I can read the error in case it occurs?

**Comment by MMasterson** [comment]
@kpavankotesh of course! You should always be to read errors, i'll patch it up for you.
